jps-plugin/src/main/kotlin/org/rust/jps/RustBuilder.kt

```diff
@@ -0,0 +1,129 @@
+package org.rust.jps
+
+
+import org.jetbrains.jps.ModuleChunk
+import org.jetbrains.jps.builders.DirtyFilesHolder
+import org.jetbrains.jps.builders.java.JavaSourceRootDescriptor
+import org.jetbrains.jps.incremental.*
+import org.jetbrains.jps.incremental.messages.BuildMessage
+import org.jetbrains.jps.incremental.messages.CompilerMessage
+import org.jetbrains.jps.model.module.JpsModule
+
+import java.io.BufferedReader
+import java.io.File
+import java.io.IOException
+import java.io.InputStreamReader
+import java.util.regex.Pattern
+
+
+class RustBuilder : ModuleLevelBuilder(BuilderCategory.TRANSLATOR) {
+
+
+    @Throws(ProjectBuildException::class, IOException::class)
+    override fun build(context: CompileContext,
+                       chunk: ModuleChunk,
+                       dirtyFilesHolder: DirtyFilesHolder<JavaSourceRootDescriptor, ModuleBuildTarget>,
+                       outputConsumer: ModuleLevelBuilder.OutputConsumer): ModuleLevelBuilder.ExitCode {
+
+        for (module in chunk.modules) {
+            if (module.moduleType !is JpsRustModuleType) {
+                continue
+            }
+
+            context.processMessage(CompilerMessage("cargo", BuildMessage.Kind.INFO, "cargo build"))
+
+            val path = getContentRootPath(module)
+
+            val processBuilder = ProcessBuilder("cargo", "build")
+            processBuilder.redirectErrorStream(true)
+            processBuilder.directory(File(path))
+            val process = processBuilder.start()
+            processOut(module, context, process)
+            try {
+                process.waitFor()
+            } catch (e: InterruptedException) {
+                e.printStackTrace()
+            }
+
+            if (process.exitValue() != 0) {
+                return ModuleLevelBuilder.ExitCode.ABORT
+            }
+        }
+
+        return ModuleLevelBuilder.ExitCode.OK
+    }
+
+    @Throws(IOException::class)
+    private fun processOut(module: JpsModule, context: CompileContext, process: Process) {
+        val processOut = collectOutput(process)
+
+        while (processOut.hasNext()) {
+            val line = processOut.next()
+
+            val matcher = Pattern.compile("(.*):(\\d+):(\\d+): (\\d+):(\\d+) error:(.*)").matcher(line)
```

Comment:
```"(.*):(\\d+):(\\d+): (\\d+):(\\d+) error:(.*)```

Is there any concise way to avoid this `\\`? 


I believe there is some extension method to do this `compile then match` in one step. There is at least `toRegex()`
```
"""(.*):(\d+):(\d+): (\d+):(\d+) error:(.*)""".toRegex()
```